### PR TITLE
Fix various JET errors

### PR DESCRIPTION
**Pull request by odow**
Part of #2268 

Now down to `═════ 70 possible errors found ═════`

https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6242312138

### CODE REVIEW

File: src/Utilities/objective_container.jl [hunk 1]
```diff
@@ -118,8 +118,20 @@ end
...
+        MOI.VariableIndex,
+        MOI.ScalarAffineFunction{T},
+        MOI.ScalarQuadraticFunction{T},
+        MOI.ScalarNonlinearFunction,
+        MOI.VectorOfVariables,
+        MOI.VectorAffineFunction{T},
+        MOI.VectorQuadraticFunction{T},
+        MOI.VectorNonlinearFunction,
+    },
+}
```
Comment (odow) [inline src/Utilities/objective_container.jl:134]: Hmm. This breaks JuMP:
https://github.com/jump-dev/JuMP.jl/blob/1ee5faee4335386435e3211b8758fba13799234e/test/test_objective.jl#L168-L180

I'm not sure whether we intended this:
```julia
julia> model = GenericModel{Float32}();

julia> @variable(model, x)
x

julia> @objective(model, Min, 2 * x + 1)
2 x + 1

julia> f = objective_function(model, GenericAffExpr{Float64,VariableRef})
2 x + 1

julia> typeof(f)
GenericAffExpr{Float64, GenericVariableRef{Float32}}

julia> g = objective_function(model)
2 x + 1

julia> typeof(g)
GenericAffExpr{Float32, GenericVariableRef{Float32}}
```

Comment (odow) [reply src/Utilities/objective_container.jl:134]: https://github.com/jump-dev/MathOptInterface.jl/actions/runs/6244194786

Comment (blegat) [reply src/Utilities/objective_container.jl:134]: No I think we meant to use `T` instead of `Float64` in the test

File: src/Utilities/model.jl [hunk 2]
```diff
@@ -34,10 +34,6 @@ function MOI.add_variable(model::AbstractModel)
     return x
 end
 
-function MOI.add_variables(model::AbstractModel, n::Integer)
```
Comment (blegat) [inline src/Utilities/model.jl:37]: Where did that one go ?

Comment (odow) [reply src/Utilities/model.jl:37]: It's the same as the default fallback

Comment (blegat) [reply src/Utilities/model.jl:37]: Ah yes, it's in `Utilities/model.jl`

**Review by blegat — APPROVED** [review]

**Pull request merged**